components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts

```diff
@@ -60,28 +60,37 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       }
       break
     case types.ON_TAB_RETRIEVED:
-      const tab: chrome.tabs.Tab = payload.tab
-      if (
-        !tab ||
-        !tab.url ||
-        tab.incognito ||
-        !tab.active ||
-        !state.walletCreated
-      ) {
-        break
-      }
+      {
+        const tab: chrome.tabs.Tab = payload.tab
+        if (
+          !tab ||
+          !tab.url ||
+          tab.incognito ||
+          !tab.active ||
+          !state.walletCreated
+        ) {
+          break
+        }
 
-      chrome.braveRewards.getPublisherData(tab.windowId, tab.url, tab.favIconUrl || '')
-      const id = getWindowId(tab.windowId)
-      let publishers: Record<string, RewardsExtension.Publisher> = state.publishers
-      if (publishers[id]) {
-        delete publishers[id]
-      }
-      state = {
-        ...state,
-        publishers
+        const id = getWindowId(tab.windowId)
+        const publishers: Record<string, RewardsExtension.Publisher> = state.publishers
+        const publisher = publishers[id]
+        chrome.braveRewards.getPublisherData(tab.windowId, tab.url, tab.favIconUrl || '')
+        if (!publisher || (publisher && publisher.tabUrl !== tab.url)) {
+          if (publisher) {
+            delete publishers[id]
+          }
+
+          publishers[id] = {
+            tabUrl: tab.url
```

Comment:
(just checking) do we want `tabUrl` always set to `tab.url` even if `!publisher` is true?

Reply:
yes so that we can track when switching tabs or reopening panel